smb3parse/util/rom.py

```diff
@@ -89,45 +123,67 @@ def tsa_data_for_object_set(self, object_set: int) -> bytearray:
 
         return self.read(tsa_start, TSA_TABLE_SIZE)
 
-    def little_endian(self, offset: int) -> int:
-        offset = self.prg_normalize(offset)
-        return little_endian(self._data[offset : offset + 2])
+    def little_endian(self, offset: AnyAddress) -> int:
+        return little_endian(self.read(offset, 2))
 
-    def write_little_endian(self, offset: int, integer: int):
-        offset = self.prg_normalize(offset)
+    def write_little_endian(self, offset: AnyAddress, integer: int):
         right_byte = (integer & 0xFF00) >> 8
         left_byte = integer & 0x00FF
 
         self.write(offset, bytes([left_byte, right_byte]))
 
-    def read(self, offset: int, length: int) -> bytearray:
+    def read(self, offset: AnyAddress, length: int) -> bytearray:
         offset = self.prg_normalize(offset)
+
+        return self._read(offset, length)
+
+    def _read(self, offset: NormalizedAddress, length: int) -> bytearray:
         return self._data[offset : offset + length]
 
-    def read_until(self, offset, delimiter):
+    def read_until(self, offset: AnyAddress, delimiter: Union[bytes, int]):
+        if isinstance(delimiter, int):
+            delimiter = bytes([delimiter])
+
         end = self.find(delimiter, offset)
 
         return self.read(offset, end - offset)
 
-    def write(self, offset: int, data: Union[bytes, int]):
+    def write(self, offset: AnyAddress, data: Union[bytes, int]):
         if isinstance(data, int):
             data = bytes([data])
 
         offset = self.prg_normalize(offset)
+
+        return self._write(offset, data)
+
+    def _write(self, offset: NormalizedAddress, data: bytes):
         self._data[offset : offset + len(data)] = data
 
-    def find(self, byte: bytes, offset: int = 0) -> int:
-        return self._data.find(byte, offset)
+    def find(self, needle: Union[bytes, int], start: AnyAddress = 0, end: AnyAddress = -1) -> int:
```

Comment:
I believe `find` should return a `NormalizedAddress`. If you don't, then you can't use any found address to read from the ROM without it trying to normalize that address again.

Reply:
You're right.